network/p2p/middleware.go

```diff
@@ -456,6 +520,13 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 
 	remotePeer := s.Conn().RemotePeer()
 
+	// check if unicast stream creation is rate limited for peer
+	if !m.unicastStreamAllowed(remotePeer) {
+		log.Warn().Msg(UnicastStreamRateLimited)
```

Comment:
please consider adding a metric along with a log like, so we can easily graph it.  It would also be awesome if we can graph it in dry-run.

Reply:
This enables us to collect metrics during dry run https://github.com/onflow/flow-go/pull/2840/commits/0efe2d3ec0035011c626c773487e5bb4dca579de waiting for this PR to get merged so I can add the line of code to collect the metric **_rate_limited_unicast_messages_count_** https://github.com/onflow/flow-go/pull/3146

Reply:
collecting metrics https://github.com/onflow/flow-go/pull/2840/commits/3c3b9a934dae846b95b27d85a1d2fbbf02f0e117